src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -101,7 +101,8 @@ const SignOrExecuteForm = ({
   const onSign = async (): Promise<string> => {
     const [connectedWallet, createdTx, provider] = assertSubmittable()
 
-    const shouldEthSign = isHardwareWallet(connectedWallet) || isSafeMobileWallet(connectedWallet)
+    const shouldEthSign =
+      isHardwareWallet(connectedWallet) || isSafeMobileWallet(connectedWallet) || isWalletConnect(connectedWallet)
```

Comment:
I suggest we create a helper for this with a brief note as to why each requires it. Would be nice to add a test case for it as well.